pkg/controller/catalogsourceconfig/reconciler.go

```diff
@@ -32,5 +33,5 @@ import (
 // Reconcile operation is expected to be idempotent so that in the event of
 // multiple invocations there would be no adverse or side effect.
 type Reconciler interface {
-	Reconcile(ctx context.Context, in *v1alpha1.CatalogSourceConfig) (out *v1alpha1.CatalogSourceConfig, nextPhase *v1alpha1.Phase, err error)
+	Reconcile(ctx context.Context, in *v1alpha1.CatalogSourceConfig, client client.Client) (out *v1alpha1.CatalogSourceConfig, nextPhase *v1alpha1.Phase, err error)
```

Comment:
I agree, we should not change the interface to accept a `client`.  `client` should be added to the implementing struct as a member variable. Because the `Reconciler` interface does not really to take into account a `client` object. All it cares about is accept an object as input and provide a result as output.

Reply:
Yep, I agree. I did this in the operatorsource package and must have forgotten to modify it in the catalogsourceconfig package when I did... I'll update it now.

Reply:
Done